--- util.c	2007-11-26 04:40:29.000000000 +0100
+++ util.c.oden	2009-10-15 18:44:12.000000000 +0200
@@ -178,7 +178,7 @@ void read_config_options() {
 	/* determine logfile path */
 	if ((res = getsetting(&mysql, "path_cactilog")) != 0 ) {
 		if (strlen(res) != 0) {
-			snprintf(set.path_logfile, SMALL_BUFSIZE, res);
+			snprintf(set.path_logfile, SMALL_BUFSIZE, "%s", res);
 		}else{
 			if (strlen(web_root) != 0) {
 				snprintf(set.path_logfile, SMALL_BUFSIZE, "%s/log/cacti.log", web_root);
@@ -495,7 +495,7 @@ void die(const char *format, ...) {
 		snprintf(flogmessage, BUFSIZE, "%s (Cactid init)", logmessage);
 	}
 
-	CACTID_LOG((flogmessage));
+	CACTID_LOG(("%s", flogmessage));
 
 	if (set.parent_fork == CACTID_PARENT) {
 		if (set.php_initialized) {
